### ISSUE TITLE
SSL issue postgres

### ISSUE THREAD

**Issue by krystian50** [cube-js/cube.js#679]
**Describe the bug**
I have some issues with SSL after upgrading the "@cubejs-backend/postgres-driver" to version `0.19.20`. Was there any change? I can't run the server in dev mode on localhost. Everything was fine before. I have no special SSL configuration.

**To Reproduce**
1. Upgrade "@cubejs-backend/postgres-driver" from `0.19.17` to `0.19.20`
2. Load any query
3. Issue `Error: Error: self signed certificate`
**Expected behavior**
Query should be performed

**Screenshots**

**Version:**
 "@cubejs-backend/postgres-driver": 0.19.20

**Additional context**

**Comment by paveltiunov** [comment]
@krystian50 Hey Krystian! Thanks for posting this! I believe we simply updated `pg` dependency. You may want to see https://node-postgres.com/announcements#2020-02-25 if you're using self signed cert.

**Comment by 0x62** [comment]
You can fix this by setting environment variable `CUBEJS_DB_SSL_REJECT_UNAUTHORIZED` to `false`.

Not sure why this wasn't listed as a breaking change, self signed certs for PostgreSQL are incredibly common.

**Comment by 0x62** [comment]
You can fix this by setting environment variable `CUBEJS_DB_SSL_REJECT_UNAUTHORIZED` to `false`.

Not sure why this wasn't listed as a breaking change for @cubejs-backend/postgres-driver, self signed certs for PostgreSQL are incredibly common.

**Comment by jcw-** [comment]
TL;DR: Setting `CUBEJS_DB_SSL_REJECT_UNAUTHORIZED` to `false` on production has security implications

FYI, while setting that to `false` may work, it's because it downgrades the security negotiation such that it will connect to a database endpoint with ANY ssl cert, including a random one provided by a man-in-the-middle (MITM) attack. Maybe this is acceptable to you, but be aware that conceptually this isn't really fixing something that is broken, it's just a workaround to avoid setting up ssl at the higher "strength".

Cube.js supports the higher strength ssl connection patterns too - see https://cube.dev/docs/connecting-to-the-database#notes-aws-rds-postgres for example, which describes setting the certificate authority to the actual certificate chain of your target (by putting the pem file / certificate chain as text into a specific env var), allowing you to set `sslmode` to `verify-full` to gain MITM protection.

Another technique is to deploy the certificate chain as a `.pem` file to your service and reference it from node's `NODE_EXTRA_CA_CERTS` env var.
